packages/block-library/src/post-navigation-link/edit.js

```diff
@@ -47,6 +60,44 @@ export default function PostNavigationLinkEdit( {
 			[ `has-text-align-${ textAlign }` ]: textAlign,
 		} ),
 	} );
+
+	// We need to know the post type in order to get all supported taxonomies.
+	const postType = useSelect(
+		// FIXME: @wordpress/block-library should not depend on @wordpress/editor.
```

Comment:
It is consistent with how some other blocks handle this store, but it feels like a temporary solution and it is part of a bigger discussion. I believe this is the main issue: https://github.com/WordPress/gutenberg/issues/32842
@gziolo

Reply:
This won't work as expected in the site editor by default. It would only work correctly if a plugin somehow enqueues `wp-editor`, but it isn't true for Gutenberg and WordPress core.

Have you tried sourcing the same information from the block context by defining: `"usesContext": [ "postType" ],` in `block.json`? Example:

https://github.com/WordPress/gutenberg/blob/bc704e140ce6844ce2dc1666bb3697801970bdc7/packages/block-library/src/post-author-biography/block.json#L14

Reply:
Thank you, I will try that and reply here.

Reply:
Let me know if there are any issues with the idea. We can always land as is and work on the general solution in the follow-up.

Reply:
_Context_ seems to be working well in both the Site Editor and post editor. But I am testing manually, not sure which edge cases I need to test and how. 🤔

Reply:
https://github.com/WordPress/gutenberg/pull/48912/commits/54a58bb9a10c287ee3cf86c6004705f25d3d4ce9 that looks great :+1: